### ISSUE TITLE
verify can't be accessed

### ISSUE THREAD

**Issue by kodemaniak** [scallop/scallop#72]
I am currently trying to define a ScallopConf object that uses some derived members. I know that I need to call `verify` to parse the definitions. All examples look roughly as follows:

```
class Cli(args: Array[String]) extends ScallopConf(args) {
  val configFile = opt[String]("config", descr = "Configuration file to use.")
  verify
  val config = configFile.get.map(f => ConfigFactory.parseFile(new  File(f))).getOrElse(ConfigFactory.empty)
}
```

However, this piece of code produces an error in my project, namely that verify can not be accessed from the `Cli` class I'm trying to define. Looking at the code it seems that verify is package private, so I tend agree with the compiler ;) Am I missing something important, or is this a bug?

**Comment by Rogach** [comment]
That's a bug - a bug in documentation, to be specific. Where did you find those examples?

Your code should probably look like:

``` scala
class Cli(args: Array[String]) extends ScallopConf(args) {
  val configFile = opt[String]("config")
  lazy val config = configFile,get.map(f => ConfigFactory.parseFile(new  File(f))).getOrElse(ConfigFactory.empty)
}
```

`verify` gets called automatically (at `}` brace). Also, note that `config` is lazy val - if it would be non-lazy, it would be evaluated before `verify` call.

**Comment by kodemaniak** [comment]
The first time I saw that in the README under "Fancy Things". Then I had a look at some tests, and there the `verify` is also  used, e.g., here: https://github.com/Rogach/scallop/blob/master/src/test/scala/ConfTest.scala under trailing args. In the tests it obviously works since they are located in the scallop package.

I assumed that using lazy vals would be a solution and I saw you mentioning it in the docs, I just wanted to make sure that this is the way to go.

Thanks!
